packages/shared/src/utils/helpers/chart-barrier-store.ts

```diff
@@ -1,19 +1,21 @@
 import { action, computed, observable, makeObservable } from 'mobx';
-import { BARRIER_LINE_STYLES, CONTRACT_SHADES, DEFAULT_SHADES } from '@deriv/shared';
-import { barriersToString } from './Helpers/barriers';
+import { BARRIER_LINE_STYLES, CONTRACT_SHADES, DEFAULT_SHADES } from '../constants';
+import { barriersToString } from './barriers';
 
 type TOnChartBarrierChange = null | ((barrier_1: string, barrier_2?: string) => void);
-type TOnChangeParams = { high: string | number; low?: string | number };
+type TOnChangeParams = { high: string | number; low?: string | number; title?: string };
 type TChartBarrierStoreOptions =
     | {
-          color: string;
+          color?: string;
+          hideBarrierLine?: boolean;
           line_style?: string;
           not_draggable?: boolean;
+          shade?: string;
       }
     | Record<string, never>;
 
 export class ChartBarrierStore {
```

Comment:
I had doubts about it too but It was used in both `core` and `trader`, and we had 2 copies of this file, which were imported into various files and used to created barrier objects by calling their constructors. Since this class is used to create a barrier object, I really wonder why someone decided to use it with mobX some time ago.